packages/fx-core/src/plugins/resource/localdebug/index.ts

```diff
@@ -326,25 +343,144 @@ export class LocalDebugPlugin implements Plugin {
+
+      const localEnvMultiProvider = new LocalEnvMultiProvider(ctx.root);
+      const frontendEnvs = includeFrontend
+        ? await localEnvMultiProvider.loadFrontendLocalEnvs(includeBackend, includeAuth)
+        : undefined;
+      const backendEnvs = includeBackend
+        ? await localEnvMultiProvider.loadBackendLocalEnvs()
+        : undefined;
+      const botEnvs = includeBot
+        ? await localEnvMultiProvider.loadBotLocalEnvs(isMigrateFromV1)
+        : undefined;
+
+      // get config for local debug
+      const clientId = ctx.localSettings?.auth?.get(LocalSettingsAuthKeys.ClientId) as string;
+      const clientSecret = ctx.localSettings?.auth?.get(
+        LocalSettingsAuthKeys.ClientSecret
+      ) as string;
+      const applicationIdUri = ctx.localSettings?.auth?.get(
+        LocalSettingsAuthKeys.ApplicationIdUris
+      ) as string;
+      const teamsAppTenantId = ctx.localSettings?.teamsApp?.get(
+        LocalSettingsTeamsAppKeys.TenantId
+      ) as string;
+      const teamsMobileDesktopAppId = TeamsClientId.MobileDesktop;
+      const teamsWebAppId = TeamsClientId.Web;
+      const localAuthEndpoint = ctx.localSettings?.auth?.get(
+        LocalSettingsAuthKeys.SimpleAuthServiceEndpoint
+      ) as string;
+      const localTabEndpoint = ctx.localSettings?.frontend?.get(
+        LocalSettingsFrontendKeys.TabEndpoint
+      ) as string;
+      const localFuncEndpoint = ctx.localSettings?.backend?.get(
+        LocalSettingsBackendKeys.FunctionEndpoint
+      ) as string;
+
+      if (includeFrontend) {
+        if (includeAuth) {
+          frontendEnvs!.teamsfxLocalEnvs[EnvKeysFrontend.TeamsFxEndpoint] = localAuthEndpoint;
+          frontendEnvs!.teamsfxLocalEnvs[
+            EnvKeysFrontend.LoginUrl
+          ] = `${localTabEndpoint}/auth-start.html`;
+          frontendEnvs!.teamsfxLocalEnvs[EnvKeysFrontend.ClientId] = clientId;
+        }
+
+        if (includeBackend) {
+          frontendEnvs!.teamsfxLocalEnvs[EnvKeysFrontend.FuncEndpoint] = localFuncEndpoint;
+          frontendEnvs!.teamsfxLocalEnvs[EnvKeysFrontend.FuncName] = ctx.projectSettings
+            ?.defaultFunctionName as string;
+
+          backendEnvs!.teamsfxLocalEnvs[EnvKeysBackend.FuncWorkerRuntime] = "node";
+          backendEnvs!.teamsfxLocalEnvs[EnvKeysBackend.ClientId] = clientId;
+          backendEnvs!.teamsfxLocalEnvs[EnvKeysBackend.ClientSecret] = clientSecret;
+          backendEnvs!.teamsfxLocalEnvs[EnvKeysBackend.AuthorityHost] =
+            "https://login.microsoftonline.com";
+          backendEnvs!.teamsfxLocalEnvs[EnvKeysBackend.TenantId] = teamsAppTenantId;
+          backendEnvs!.teamsfxLocalEnvs[EnvKeysBackend.ApiEndpoint] = localFuncEndpoint;
+          backendEnvs!.teamsfxLocalEnvs[EnvKeysBackend.ApplicationIdUri] = applicationIdUri;
+          backendEnvs!.teamsfxLocalEnvs[EnvKeysBackend.AllowedAppIds] = [
+            teamsMobileDesktopAppId,
+            teamsWebAppId,
+          ].join(";");
+        }
```

Comment:
The local env generation is quite similar to `getLocalDebugEnvs`.
As we will support env for local debug, can we consolidate this part? And we can leverage the new `.env.teamsfx.local` to pass the local debug env to VS Code extension.